### PR TITLE
Adds a space manager

### PR DESCRIPTION
**Pull request by Crazylemon64**
This lets you add level attributes freely, and will allow you to rebuild the transitions of specific levels with only a smidgen of extra code.
This doesn't do any of the work needed to make every system check the space manager, however.

### CODE REVIEW

**Comment by CrAzYPiLoT-SS13** [comment]
Thank you, thank you, thank you.

File: _maps/__MAP_DEFINES.dm [hunk 1]
```diff
@@ -28,4 +52,19 @@ Last space-z level = empty
...
+
+#define DECLARE_LEVEL(NAME,LINKS,TRAITS) list("name" = NAME, "linkage" = LINKS, "attributes" = TRAITS)
+
+
+#define DEFAULT_MAP_TRANSITION_CONFIG list(\
+DECLARE_LEVEL(MAIN_STATION,CROSSLINKED,list(STATION_LEVEL,STATION_CONTACT,REACHABLE,AI_OK)),\
+DECLARE_LEVEL(CENTCOMM,SELFLOOPING,list(ADMIN_LEVEL, BLOCK_TELEPORT, IMPEDES_MAGIC)),\
+DECLARE_LEVEL(EMPTY_AREA_1,CROSSLINKED,list(REACHABLE)),\
+DECLARE_LEVEL(EMPTY_AREA_2,CROSSLINKED,list(REACHABLE)),\
+DECLARE_LEVEL(MINING,SELFLOOPING,list(REACHABLE, STATION_CONTACT, HAS_WEATHER, ORE_LEVEL, AI_OK)),\
```
Comment (Crazylemon64) [inline _maps/__MAP_DEFINES.dm]: Should mining have the "Reachable" attribute?

Comment (CrAzYPiLoT-SS13) [inline _maps/__MAP_DEFINES.dm]: We work by weird principles regarding that. We don't actually have mining. We will most likely need to adapt this heavily.

**Comment by CrAzYPiLoT-SS13** [comment]
Travis doesn't approve

**Comment by Crazylemon64** [comment]
By runtimes, I mean compile errors, bah

**Comment by Crazylemon64** [comment]
This DOES add a system to suspend init for when loading a sophisticated map such as a space station, but I haven't integrated it into the systems that would use it.

**Comment by monster860** [comment]
Have you integrated this with the code calling the maploader in code/controllers/subsystem/mapping.dm?

**Comment by Crazylemon64** [comment]
I have, but only in the part that sets up map transitions, and where it would make sense to put the space manager's initializiation.

**Comment by Crazylemon64** [comment]
Also, I'm noticing that some of your code has `initialize` called in the `New` of the very same object - this seems strange to me - when exactly is `initialize` normally called for your objects?

**Comment by monster860** [comment]
Not sure about the initialize thing.

**Pull request merged**